### PR TITLE
Redesign chat layout for improved text selection experience

### PR DESCRIPTION
**Pull request by pkukielka**
Fixes https://linear.app/sourcegraph/issue/CODY-6024/generated-code-blocks-in-intellij-dont-autoscroll-making-it-difficult

# Fix text selection UX by redesigning chat layout with fixed input box

## Problem

Users were experiencing difficulty selecting large chunks of text in chat conversations. When trying to select text that spans multiple screens by clicking and dragging, the selection would not expand when the mouse cursor moved under the input box, which was overlapping the bottom portion of the chat content.

## Solution

Redesigned the chat layout to separate the input box from the scrollable chat content:

- **Fixed input box** at the bottom of the screen during active conversations
- **Scrollable chat content** area above the input box
- **Smooth fade transition** where content meets the input box
- **Proper welcome message flow** for new conversations

### 1. **Layout Restructure (`Transcript.tsx`)**
- Split interactions into `chatInteractions` (scrollable area) and `inputInteraction` (fixed bottom)
- Added logic to determine when input should be fixed vs. in scrollable area:
  - **First interaction**: Input appears at top in scrollable area
  - **Subsequent interactions**: Input fixed at bottom with chat content scrollable above
- Implemented CSS mask fade effect for smooth visual transition between content and input

image

### 2. **Welcome Message Positioning**
- Keep original layout: Input box → PromptList → LastConversation (at bottom)
- Split welcome content into `welcomeContentTop` and `welcomeContentBottom` for proper positioning
- Maintains original UX while fixing text selection issues

### 3. **Visual Improvements**
- Added subtle fade effect using CSS `maskImage` with CSS variable support (`--cody-fade-height`)
- Removed harsh components border separator for smoother visual flow
- Consistent padding (`tw-px-8`) across all components for width alignment
- Proper background styling for fixed input area

### 4. **Code Quality Improvements**
- **Reduced duplication**: Extracted `commonTranscriptProps` for shared component props
- **Performance**: Memoized `fadeStyles` object to prevent unnecessary re-renders
- **Maintainability**: Added helper functions and improved conditional logic
- **Error handling**: Added safety checks for edge cases

## Test plan

1. Start longer chat conversation
2. Scroll to the beginning of the conversation
3. Click on the text and drag mouse cursor to the bottom of the chat to expand the selection
4. Expand the selection over the multiple screens of text

### CODE REVIEW

File: vscode/webviews/chat/Transcript.tsx [hunk 1]
```diff
@@ -175,57 +209,101 @@ export const Transcript: FC<TranscriptProps> = props => {
...
+                            editorRef={undefined} // Input editor is handled separately
+                        />
+                    ))}
+                    {/* Show loading dots after the last message in scrollable area */}
+                    {messageInProgress && chatInteractions.length > 0 && <LoadingDots />}
+                    {/* Show welcome content bottom when starting a new conversation */}
+                    {shouldShowWelcomeContent && welcomeContentBottom}
+                </LastEditorContext.Provider>
+            </div>
+
```
Comment (abeatrix) [inline vscode/webviews/chat/Transcript.tsx:276]:
```suggestion
            <ScrollbarMarkers />
```

we should probablay move the marker above the input box

File: vscode/webviews/chat/Transcript.tsx [hunk 2]
```diff
@@ -175,57 +209,101 @@ export const Transcript: FC<TranscriptProps> = props => {
...
                             }
                         />
-                    ))}
-                </LastEditorContext.Provider>
-            </div>
+                    </LastEditorContext.Provider>
+                </div>
+            )}
+
             <ScrollbarMarkers />
```
Comment (abeatrix) [inline vscode/webviews/chat/Transcript.tsx:305]:
```suggestion
```

**Review by abeatrix — APPROVED** [review]
Works as described. Auto-scroll works.

**Review by ichim-david — APPROVED** [review]
Confirming that it works now, thank you @pkukielka for this work
